packages/dashboard/src/components/iot-dashboard/iot-dashboard.tsx

```diff
@@ -0,0 +1,271 @@
+
+  @State() currWidth: number;
+  @Element() el!: HTMLElement;
+
+  private resizer: ResizeObserver;
+
+  componentWillLoad() {
+    this.currDashboardConfiguration = this.dashboardConfiguration;
+
+    /**
+     * Creates a listener for elements dimensions changing.
+     * This allows us to dynamically set the widget dimensions.
+     */
+    this.resizer = new ResizeObserver((entries) => {
+      entries.forEach((entry) => {
+        /** Update Size */
+        const { width } = entry.contentRect;
+        this.currWidth = width;
+      });
+    });
+
+    this.currWidth = this.width;
+  }
+
+  componentDidLoad() {
+    this.resizer.observe(this.el.firstElementChild as Element);
+  }
+
+  @Watch('dashboardConfiguration')
+  watchDashboardConfiguration(newDashboardConfiguration: DashboardConfiguration) {
+    this.currDashboardConfiguration = newDashboardConfiguration;
+  }
+
+  resize = (position: Position, side: WidgetSide, widgetId: string) => {
+    this.resizingInfo = { side, widgetId };
+    this.previousPosition = position;
+  };
+
+  onStart({ x, y }: Position) {
+    const intersectedWidgetIds = getSelectedWidgetIds({
+      selectedRect: { x, y, width: 1, height: 1 },
+      dashboardConfiguration: this.currDashboardConfiguration,
+      cellSize: this.actualCellSize(),
+    });
+
+    if (intersectedWidgetIds.length === 0) {
+      // Clicking on a selected widget
+      this.start = { x, y };
+      this.finishedSelecting = false;
+    } else {
+      // Begin moving widgets selected
+      this.previousPosition = { x, y };
+      if (this.selectedWidgetIds.length === 0) {
+        this.selectedWidgetIds = intersectedWidgetIds;
+      }
+    }
+  }
+
+  setDashboardConfiguration(dashboardConfiguration: DashboardConfiguration) {
+    this.currDashboardConfiguration = dashboardConfiguration;
+    this.onDashboardConfigurationChange(this.currDashboardConfiguration);
+  }
+
+  /**
+   * Moves one or more selected widgets
+   */
+  moveWidgets(position: Position) {
+    this.setDashboardConfiguration(
+      getMovedDashboardConfiguration({
+        dashboardConfiguration: this.currDashboardConfiguration,
+        position,
+        previousPosition: this.previousPosition,
+        selectedWidgetIds: this.selectedWidgetIds,
+        cellSize: this.actualCellSize(),
+      })
+    );
+  }
+
+  onMove({ x, y }: Position) {
+    if (this.previousPosition) {
+      /** is moving widgets */
+      this.moveWidgets({ x, y });
+      this.previousPosition = { x, y };
+    } else if (!this.finishedSelecting) {
+      /** is selecting */
+      this.end = { x, y };
+      this.setSelectedWidgets();
+    }
+  }
+
+  onEnd({ x, y }: Position) {
+    this.previousPosition = undefined;
+    this.end = { x, y };
+    this.finishedSelecting = true;
+    this.setSelectedWidgets();
+    this.start = undefined;
+    this.end = undefined;
+
+    this.currDashboardConfiguration = this.currDashboardConfiguration.map((widget) => ({
```

Comment:
Shouldn't we use `setDashboardConfiguration` here?

Reply:
updated to utilize setDashboardConfiguration